src/hooks/template/useQuestions.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+
+import { GetQuestionsQuery, QuestionsFilter } from 'generated/sdk';
+import { useDataApi } from 'hooks/common/useDataApi';
+
+export type QuestionWithUsage = GetQuestionsQuery['questions'][number];
+
+export function useQuestions(filter?: QuestionsFilter) {
+  const [questionsFilter, setQuestionsFilter] = useState(filter);
+  const [questions, setQuestions] = useState<QuestionWithUsage[]>([]);
+  const [loadingQuestions, setLoadingQuestions] = useState(true);
+
+  const api = useDataApi();
+
+  useEffect(() => {
+    setLoadingQuestions(true);
+    api()
+      .getQuestions({ filter: questionsFilter })
+      .then((data) => {
```

Comment:
Please add a check to see if the component is unmounted or not.

Reply:
Yes I think we should do that to all hooks that are using useEffect to prevent those console warnings.